kibana-reports/server/routes/lib/createReport.ts

```diff
@@ -102,13 +105,18 @@ export const createReport = async (
           }
         });
       }
-      createReportResult = await createVisualReport(
-        reportParams,
-        completeQueryUrl,
-        logger,
-        cookieObject,
-        timezone
-      );
+      const [value, release] = await semaphore.acquire();
```

Comment:
what if the acquire failed/timeout? is there retry avaliable?

Reply:
I think retrying semaphore not required. we can just increase timeout to required value.

Reply:
If request fails or timeout, client side would get an error. I think increasing timeout is better than client auto retry. I can improve the error message to indicate it's a timeout on UI